### PR TITLE
Add `CHANGELOG` for `2.0.0-beta.13` release

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

Adds `CHANGELOG` for `2.0.0-beta.13` release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,32 @@
...
 Mapbox welcomes participation and contributions from everyone.
 
+## Mapbox Navigation SDK 2.0.0-beta.13 - June 4, 2021
+
+For details on how v2 differs from v1 and guidance on migrating from v1 of the Mapbox Navigation SDK for Android to the v2 public preview, see [2.0 Navigation SDK Migration Guide](https://github.com/mapbox/mapbox-navigation-android/wiki/2.0-Navigation-SDK-Migration-Guide).
+
+### Changelog
+#### Bug fixes and improvements
+* Removed `getRouteGeometryWithBuffer` API from `RouteProgress`. [#4478](https://github.com/mapbox/mapbox-navigation-android/pull/4478)
+* Removed the `updatePrimaryRoute` method from the `MapboxRouteLineApi` class. It was replaced with an extension function `RouteLineUtils::updatePrimaryRoute` which reorders a collections of `DirectionsRoute` objects. [#4467](https://github.com/mapbox/mapbox-navigation-android/pull/4467)
```
Comment (LukasPaczos) [inline CHANGELOG.md:12]: The `updatePrimaryRoute` didn't ultimately make it into the API.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+* Recreate Navigator to switch tiles version. [#4451](https://github.com/mapbox/mapbox-navigation-android/pull/4451)
```
Comment (LukasPaczos) [inline CHANGELOG.md:13]: Could you elaborate here a little bit? Let's at least mention what this means and link to the interface.

Comment (Guardiola31337) [reply CHANGELOG.md:13]: How about?

> Added `NavigationVersionSwitchObserver` which enables listening to navigation tiles version switch.

File: CHANGELOG.md (hunk 1 above)
Comment (Guardiola31337) [reply CHANGELOG.md:12]: Updating to 

> Removed the updatePrimaryRoute method from the MapboxRouteLineApi class. Instead, MapboxRouteLineApi::setRoutes should be used. The first DirectionsRoute in the collection will be considered the primary route.

cc @cafesilencio

**Comment by Guardiola31337** [comment]
I've just addressed the comments. Ready for another round of 👀

**Review by cafesilencio — APPROVED** [review]

**Pull request merged**